.gitignore

```diff
@@ -8,3 +8,4 @@ build/
 packages
 .idea/
 
+pubspec.lock
```

Comment:
It doesn't explicitly. But if someone else on your dev team does an update it changes the repo.

So the question is: is this an artifact that represents the "state" of the project.

For an app: likely.
For a library: likely not.